skimage/exposure/exposure.py

```diff
@@ -266,6 +266,39 @@ def intensity_range(image, range_values='image', clip_negative=False):
     return i_min, i_max
 
 
+def _output_dtype(out_range):
+    """Determine the output dtype for rescale_intensity.
+
+    The dtype is determined according to the following rules:
+    - if ``out_range`` is a dtype, that is the output dtype.
+    - if ``out_range`` is a dtype string, that is the dtype used, unless it
+      is not a NumPy data type (e.g. 'uint12' for 12-bit unsigned integers),
+      in which case the data type that can contain it will be used
+      (e.g. uint16 in this case).
+    - if ``out_range`` is a pair of values, the output data type will be
+      float.
+
+    Parameters
+    ----------
+    out_dtype : type
+        The data type appropriate for the desired output.
+    """
+    if type(out_range) in [list, tuple, np.ndarray]:
+        # pair of values: always return float.
+        return np.float_
+    if type(out_range) == type:
+        # already a type: return it
+        return out_range
+    if out_range in DTYPE_RANGE:
+        # string key in DTYPE_RANGE dictionary
+        try:
+            # if it's a canonical numpy dtype, convert
+            return np.dtype(out_range).type
+        except TypeError:  # uint10, uint12, uint14
+            # otherwise, return uint16
+            return np.uint16
+
```

Comment:
add an else here to raise a `TypeError` for example in the case of typos (`flat` instead of `float`). `output_dtype` should be None if none of the ifs is True, and `np.array([1, 2, 3], dtype=None)` does not error...

Reply:
I understand and support the first part of your suggestion. 

But why should `output_dtype` be `None`. The way I understand the new behavior is 
- `out_range` is an actual range -> output dtype is `float64`
- `out_range` is already a dtype (corresponding to the input dtype) -> use same type for output

`None` would just result in the returned output being of dtype `float64` wouldn't it?

Reply:
@lagru probably what I wrote was not clear, please tell me if it's clearer now. The function should never return None, it should return a valid value or error out.

Reply:
Ah, then we are in agreement. :+1: